src/main/java/dev/roryclaasen/vcsparser/measures/ComputeNumAuthorsMetric.java

```diff
@@ -0,0 +1,148 @@
+				.setInputMetrics(ObjectArrays.concat(new String[] { authorsData }, numChanges, String.class))
+				.setOutputMetrics(ObjectArrays.concat(numAuthors, numAuthors10Perc, String.class))
+				.build();
+	}
+
+	@Override
+	public void compute(MeasureComputerContext context) {
+		for (int i = 0; i < numAuthors10Perc.length; i++) {
+			switch (context.getComponent().getType()) {
+			case PROJECT:
+			case MODULE:
+			case DIRECTORY:
+				computeChildMeasure(context, numAuthors[i], numAuthors10Perc[i], numChanges[i]);
+				break;
+			case FILE:
+				computeFileMeasure(context, authorsData, numAuthors[i], numAuthors10Perc[i], numChanges[i]);
+			default:
+				break;
+			}
+		}
+	}
+
+	protected void computeChildMeasure(MeasureComputerContext context, String numAuthorsKey, String numAuthors10PercKey, String numChangesKey) {
+		String currentKey = context.getComponent().getKey();
+		String projectKey = currentKey.split(":", 2)[0];
+
+		if (!authorsCache.containsKey(projectKey))
+			return;
+
+		Map<String, List<AuthorData>> projectCache = authorsCache.get(projectKey);
+
+		Date date = getMetricDateFromKey(numAuthorsKey).getDate();
+		Map<String, Integer> numChangesDict = new HashMap<String, Integer>();
+		for (Entry<String, List<AuthorData>> entry : projectCache.entrySet()) {
+			if (!entry.getKey().startsWith(currentKey))
+				continue;
+
+			numChangesDict = getAuthorNumChangesAfterDateDict(entry.getValue(), date, numChangesDict);
+		}
+
+		computeNumAuthors(context, numAuthorsKey, numChangesDict.values());
+		computeNumAuthorsOver10Perc(context, numChangesKey, numAuthors10PercKey, numChangesDict.values());
+	}
+
+	protected void computeFileMeasure(MeasureComputerContext context, String authorsDataKey, String numAuthorsKey, String numAuthors10PercKey, String numChangesKey) {
+		String currentKey = context.getComponent().getKey();
+		String projectKey = currentKey.split(":", 2)[0];
+		
+		Measure authorsDataMeasure = context.getMeasure(authorsDataKey);
+
+		if (authorsDataMeasure == null)
+			return;
+
+		List<AuthorData> authorDataList = authorUtils.jsonStringArrayToAuthorDataList(authorsDataMeasure.getStringValue());
+		if (authorDataList.size() == 0)
+			return;
+
+		if (!authorsCache.containsKey(projectKey))
+			authorsCache.put(projectKey, new HashMap<String, List<AuthorData>>());
+		
+		authorsCache.get(projectKey).put(currentKey, authorDataList);
+
+		Collection<Integer> authorNumChanges = getAuthorNumChangesAfterDateDict(authorDataList, getMetricDateFromKey(numAuthorsKey).getDate(), new HashMap<String, Integer>()).values();
+
+		computeNumAuthors(context, numAuthorsKey, authorNumChanges);
+		computeNumAuthorsOver10Perc(context, numChangesKey, numAuthors10PercKey, authorNumChanges);
+	}
+
+	protected Map<String, Integer> getAuthorNumChangesAfterDateDict(Collection<AuthorData> authorDataList, Date dateFrom, Map<String, Integer> currentMap) {
+		List<Author> authorList = authorUtils.getAuthorListAfterDate(authorDataList, dateFrom);

```

Comment:
I think it may be an issue here. When we aggregate, we ignore the changes from the current date. We always do 7 days counting from yesterday.
Let's assume you want unique authors in the last 7 days and you run the analysis on day 8. What happened between midnight and the time of analysis of day 8 is ignored. We compute midnight of day 01 to midnight of day 8 (which doesn't consider the changes in day 8).

I'm not sure if the code considers this, since it only considers dateFrom.

Reply:
`dateFrom` is set by [`MetricUtils.getMetricDateFromKey(String)`](https://github.com/roryclaasen/sonar-vcsparser-plugin/blob/authors/src/main/java/dev/roryclaasen/vcsparser/metrics/MetricUtils.java#L12) which in turn gets its date from [`MetricDate`](https://github.com/roryclaasen/sonar-vcsparser-plugin/tree/authors/src/main/java/dev/roryclaasen/vcsparser/metrics).  
Bellow is how the dates are set up, (using the same as the [vcsparser aggregators](https://github.com/ericlemes/vcsparser/blob/master/src/vcsparser.core/MeasureConverterListBuilder.cs)).

```java
YEAR_1("_1y", " (1 year)", MinusDate(1, 0, 1)),
MONTH_6("_6m", " (6 months)", MinusDate(0, 6, 1)),
MONTH_3("_3m", " (3 months)", MinusDate(0, 3, 1)),
DAY_30("_30d", " (30 days)", MinusDate(0, 0, 31)),
DAY_7("_7d", " (7 days)", MinusDate(0, 0, 8)),
DAY_1("_1d", " (1 days)", MinusDate(0, 0, 1));

private static Date MinusDate(int year, int month, int day) { 
	Date today = new Date();
	today = DateUtils.addDays(today, -day);
	today = DateUtils.addMonths(today, -month);
	today = DateUtils.addYears(today, -year);
	return today;
}
```

It might be worth making sure that the date is zero hrs/min/sec? I'm not entirely sure how this will play out

Reply:
Right. So, good to go then ;-)